### PR TITLE
Fix connections between client-api and as servers

### PR DESCRIPTION
**Pull request by corocn**
Connection between client-api and AS containers failed.
I fixed asTransactionURL in routesController.
This PR and https://github.com/securekey/oauth-xyz-nodejs/pull/4 will make all servers launching and processing correctly with docker-compose.

### CODE REVIEW

**Comment by mavarley** [comment]
Thank-you, we will review tomorrow and merge.

**Comment by haardikk21** [comment]
This was the old URL we were using. @jricher you changed it to the internal docker host URL. Can you weigh in on this change?

**Comment by jricher** [comment]
Sorry about that, that change was done during testing against the Java implementation. It wasn't supposed to be pushed, so putting it back is fine. That said, the URL should be an external one, right? So it's `localhost:3000` and not `as:3000` with the link, since the transaction endpoint is intended to be exposed.

**Comment by haardikk21** [comment]
> Sorry about that, that change was done during testing against the Java implementation. It wasn't supposed to be pushed, so putting it back is fine. That said, the URL should be an external one, right? So it's `localhost:3000` and not `as:3000` with the link, since the transaction endpoint is intended to be exposed.

In this specific scenario, since the included client is the only one talking to the server, the internal URL would still work. Also I believe the ports are being exposed in the docker compose so the localhost URL will also work.

For consistency purposes I think we should switch to using localhost. @corocn can you update it accordingly and I will merge then.

**Comment by jricher** [comment]
While a local `link` would technically work with this, it is not the correct style of connection that this application should be using. The client and AS should be separate entities entirely, and the fact that they're packaged together in one docker container is a side effect, not a feature.

File: client/api/src/controllers/routesController.ts [hunk 1]
```diff
@@ -13,8 +13,8 @@ import {
 } from '../models/pendingTransaction';
 import * as jose from 'node-jose';
 
-//const asTransactionURL = 'http://as:3000/transaction';
-const asTransactionURL = 'http://host.docker.internal:9834/api/as/transaction';
+const asTransactionURL = 'http://as:3000/transaction';
```
Comment (jricher) [inline client/api/src/controllers/routesController.ts]:
```suggestion
const asTransactionURL = 'http://localhost:3000/transaction';
```

File: docker-compose.yml [hunk 2]
```diff
@@ -30,6 +30,7 @@ services:
       - DEBUG
     links:
       - mongo-client
+      - as
```
Comment (jricher) [inline docker-compose.yml]:
```suggestion

```

**Review by jricher — CHANGES_REQUESTED** [review]

**Comment by corocn** [comment]
> The client and AS should be separate entities entirely, and the fact that they're packaged together in one docker container is a side effect, not a feature

I got it and updated my pull requests using localhost:3000.
Thank you for your review!

**Review by jricher — APPROVED** [review]

**Comment by GrlPrgrmr** [comment]
I cloned this repo and tried to bring local env up, but i am getting errors when running docker-compose up , I am seeing 97 errors in my terminal. Can someone please help me with running my local instance of this repo. I am intending to contribute to this project.

**Pull request merged**